### PR TITLE
Blog-Page & UploadThing

### PR DESCRIPTION
**Pull request by neilnicky**

### CODE REVIEW

**Comment by muadpn** [comment]
add in alt tag for image uploads for SEO optimisation contents.

File: src/app/(user)/blog/[blogSlug]/page.tsx
Comment (muadpn) [inline src/app/(user)/blog/[blogSlug]/page.tsx:1]: fix hard coding package.

File: prisma/schema.prisma
Comment (muadpn) [inline prisma/schema.prisma:1]:
```
 -  image      Image?     @relation(fields: [imageId], references: [id])
 +  image      Image     @relation(fields: [imageId], references: [id])
 ```

File: src/app/api/uploadthing/core.ts
Comment (muadpn) [inline src/app/api/uploadthing/core.ts:1]: remove unnecessary downloads. 
```
 const imageId = metadata;
 const res = await fetch(file.url);
```

File: src/app/(admin)/admin/image-uploader/page.tsx
Comment (muadpn) [inline src/app/(admin)/admin/image-uploader/page.tsx:1]: Change Alert to toast notification.

File: src/db/data/dto/blog.ts
Comment (muadpn) [inline src/db/data/dto/blog.ts:1]: - Please verify the data recieved in `getBlogPosts()` ,`getRecentPosts()`
- use errorLogger and return something.

File: src/db/data/dto/package.ts
Comment (muadpn) [inline src/db/data/dto/package.ts:1]: Review `getPackageForBlog` similiar API is available or not

File: src/lib/validators/BlogFormValidators.tsx
Comment (muadpn) [inline src/lib/validators/BlogFormValidators.tsx:1]: Please set a valid minimum and maximum length for all contents

File: src/server/routers/admin/blog.ts
Comment (muadpn) [inline src/server/routers/admin/blog.ts:1]: Check whether the slug is already exists, and a valid slug format, and check security measures [!IMPORTANT]

File: src/server/routers/admin/schedule.ts
Comment (muadpn) [inline src/server/routers/admin/schedule.ts:1]: Remove `import { blog } from "./blog";`

File: src/server/uploadthing.ts
Comment (muadpn) [inline src/server/uploadthing.ts:1]: replace / delete this API.

File: src/utils/uploadthing.ts
Comment (muadpn) [inline src/utils/uploadthing.ts:1]: Refactor and replace this file into seperate files

File: prisma/schema.prisma
Comment (neilnicky) [reply prisma/schema.prisma:1]: Hi,

this issue has been fixed!

File: src/app/api/uploadthing/core.ts
Comment (neilnicky) [reply src/app/api/uploadthing/core.ts:1]: Hi,

I’ve refactored the code to remove the unnecessary fetching call in `uploadthing`. Please review the changes and let me know if there are any other issues.

Thanks!

File: src/server/uploadthing.ts
Comment (neilnicky) [reply src/server/uploadthing.ts:1]: Hi, 

It has been deleted for now.

Thanks.

File: src/app/(admin)/admin/image-uploader/page.tsx
Comment (neilnicky) [reply src/app/(admin)/admin/image-uploader/page.tsx:1]: Hi,

Alert has been changed to toast. 
Please review the changes and let me know if anything else is needed.  


Thanks!

File: src/lib/validators/BlogFormValidators.tsx
Comment (neilnicky) [reply src/lib/validators/BlogFormValidators.tsx:1]: Hi,

I've updated the validation rules to set valid minimum and maximum lengths for all content fields in BlogFormValidators. Each field now meets the required constraints.

Thanks for pointing this out!

File: src/server/routers/admin/blog.ts
Comment (neilnicky) [reply src/server/routers/admin/blog.ts:1]: Hi,

I’ve completed the updates:

Checked for existing slugs
Validated slug format
Added security measures
Let me know if anything else is needed.

Thanks!

Best,
[Your Name]

File: src/server/routers/admin/schedule.ts
Comment (neilnicky) [reply src/server/routers/admin/schedule.ts:1]: Hi,

I've removed the import { blog } from "./blog"; as requested.

Thanks for pointing this out!

File: src/utils/uploadthing.ts
Comment (neilnicky) [reply src/utils/uploadthing.ts:1]: Hi,

It has been moved to components/admin/uploadthing

Thanks.

File: src/db/data/dto/blog.ts
Comment (neilnicky) [reply src/db/data/dto/blog.ts:1]: Hi,

I've verified the data received in `getBlogPosts()` ,`getRecentPosts()` and added errorLogger.

Thanks for pointing this out!

File: src/db/data/dto/package.ts
Comment (neilnicky) [reply src/db/data/dto/package.ts:1]: Hi,

Reviewed similar APIs, but my API call is designed to fetch only the necessary data.

Thanks for pointing this out!

**Pull request merged**